breathecode/authenticate/views.py

```diff
@@ -502,8 +499,16 @@ def get_github_token(request):
 
     logger.debug("Redirecting to github")
     logger.debug(params)
-
-    redirect = f'https://github.com/login/oauth/authorize?{urlencode(params)}'
+    if user_id:
+        user = User.objects.filter(id=user_id).first()
+        if not user:
+            raise ValidationException('User was not found, please input different user',
+                code=404, slug='user-not-found')
+        print(user)
+        arg = {'user' : user_id}
+        redirect = f'https://github.com/login/oauth/authorize?{urlencode(params)}&{urlencode(arg)}'
+    else:
+        redirect = f'https://github.com/login/oauth/authorize?{urlencode(params)}'
```

Comment:
This line is the same that 509, should be better `if user and not User.objects.filter(id=user_id).exists()` and have just one `redirect` line, and `if user_id` you should use `params['user'] = user_id`